pkg/loki/modules.go

```diff
@@ -510,6 +516,29 @@ func (t *Loki) initQueryFrontendTripperware() (_ services.Service, err error) {
 	return services.NewIdleService(nil, nil), nil
 }
 
+func (t *Loki) frontendGenerationNumberGetter() (generationnumber.CacheGenClient, error) {
```

Comment:
could we just name this `cacheGenClient()` or `getCacheGenClient()`? we usually don't use `thingGetter` for naming in Go IME